### ISSUE TITLE
Enhance the automatic naming of string variables upon loading PDB

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#4978]
**Version and Platform (required):**
- Binary Ninja Version: 3.6.4802-dev, 87e27dc3
- OS: Windows
- OS Version: 11
- CPU Architecture: x86_64

**Bug Description:**
When loading a binary with a PDB file, BN employs unconventional variable names for string variables. These names could be enhanced for improved readability, as the current naming often blurs the distinction between variable names and variable types, leading to confusion. Additionally, this behavior makes it more challenging to copy code from the decompiler to plain C code. 

<img width="695" src="https://github.com/Vector35/binaryninja-api/assets/109607896/1dc78578-3362-4f70-b893-7cd94fb963cb">

**Steps To Reproduce:**
1. Load a binary that have a PDB file
2. Examine various string variables within the BNDB

**Expected Behavior:**
I expect BN to give those variables more simpler names. For the example above it could be sTotalAnimals. Essentially, BN could remove any characters from the variable names except for digits and alphabet letters, resulting in clearer naming.

**Comment by psifertex** [comment]
I agree it should be improved either way but I have a strong suspicion this is exactly the name that is in the PDB. And it might be tricky trying to decide when to ignore the included names and when not to.

**Comment by CouleeApps** [comment]
The official demangler calls them all `string' with no detail of the actual name. Adding the contents to the name was a choice I made when writing the demangler. We could attempt to filter the result with underscores and removing non alphanum characters, but I felt that just representing the contents directly was fine considering we're not targeting code that is directly compatible with C.
